pkgs/development/python-modules/psycopg/c.nix

```diff
@@ -0,0 +1,34 @@
+{ lib
+, buildPythonPackage
+, pythonOlder
+, psycopg
+, setuptools
+, cython_3
+, postgresql
+, pytestCheckHook
+}:
+
+buildPythonPackage {
+  pname = "psycopg-c";
+  inherit (psycopg) version;
+  format = "pyproject";
+
+  inherit (psycopg) src;
```

Comment:
Worth also inheriting any `patches` psycopg has? Best to avoid using a mismatch of source and would be good to get any security patches we hurriedly add, forgetting about the existence of these leaf packages.

Reply:
Ah yes. Patching is a bit messy. Patches are best applied to the base repo, while `sourceRoot` needs to go one directory down for each project.

Reply:
Ugh forgot patching happens relative to `sourceRoot`.